### PR TITLE
added all necessary functions

### PR DESCRIPTION
**Pull request by KrasavaStep**

### CODE REVIEW

File: app/src/main/java/com/example/lessson17/MainActivity.kt [hunk 1]
```diff
@@ -1,11 +1,124 @@
...
 import androidx.appcompat.app.AppCompatActivity
+import com.example.lessson17.databinding.ActivityMainBinding
+import java.time.LocalDateTime
+import java.time.format.DateTimeFormatter
+import kotlin.random.Random
 
 class MainActivity : AppCompatActivity() {
+    lateinit var binding: ActivityMainBinding
+
+    @RequiresApi(Build.VERSION_CODES.O)
```
Comment (vshat-tms) [inline app/src/main/java/com/example/lessson17/MainActivity.kt]: ?

File: app/src/main/java/com/example/lessson17/MainActivity.kt [hunk 2]
```diff
@@ -1,11 +1,124 @@
...
+                activity.setBackgroundColor(Color.BLUE)
+            }
+
+            buttonDevice.setOnClickListener {
+                info.text = Build.BRAND + " " + Build.MODEL
+            }
+
+            buttonTime.setOnClickListener {
+                val now = LocalDateTime.now()
+                val formatter = DateTimeFormatter.ofPattern("hh:mm")
```
Comment (vshat-tms) [inline app/src/main/java/com/example/lessson17/MainActivity.kt]: move to companion object const

File: app/src/main/java/com/example/lessson17/MainActivity.kt [hunk 3]
```diff
@@ -1,11 +1,124 @@
...
+                    Toast.LENGTH_SHORT).show()
+            }
+
+            buttonCat.setOnClickListener {
+                image.rotation = 0F
+                image.setImageResource(imageList.get(0))
+            }
+            buttonDog.setOnClickListener {
+                image.rotation = 0F
+                image.setImageResource(imageList.get(1))
```
Comment (vshat-tms) [inline app/src/main/java/com/example/lessson17/MainActivity.kt]: []

File: app/src/main/java/com/example/lessson17/MainActivity.kt [hunk 4]
```diff
@@ -1,11 +1,124 @@
...
+            }
+            buttonParrot.setOnClickListener {
+                image.rotation = 0F
+                image.setImageResource(imageList.get(2))
+            }
+            buttonRndAnimal.setOnClickListener {
+                var randomDigit = Random.nextInt(100)
+                image.rotation = 0F
+                when {
+                    randomDigit % 2 == 0 -> image.setImageResource(imageList[0])
```
Comment (vshat-tms) [inline app/src/main/java/com/example/lessson17/MainActivity.kt]: переделать

File: app/src/main/java/com/example/lessson17/MainActivity.kt [hunk 5]
```diff
@@ -1,11 +1,124 @@
...
+                    randomDigit % 5 != 0 -> image.setImageResource(imageList[1])
+                    randomDigit % 3 == 0 -> image.setImageResource(imageList[2])
+                }
+            }
+
+            image.setOnClickListener {
+                image.rotation = 90F
```
Comment (vshat-tms) [inline app/src/main/java/com/example/lessson17/MainActivity.kt]: ?

File: app/src/main/java/com/example/lessson17/MainActivity.kt [hunk 6]
```diff
@@ -1,11 +1,138 @@
...
+    private var currentImgRes = R.drawable.cat
+    private var counter = 0
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
+        binding = ActivityMainBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+
+
```
Comment (AverburhSiarhei) [inline app/src/main/java/com/example/lessson17/MainActivity.kt]: лишние пустые строки

File: app/src/main/java/com/example/lessson17/MainActivity.kt [hunk 7]
```diff
@@ -1,11 +1,138 @@
...
+        currentImgRes = IMAGE_LIST[imgResId]
+        img.setImageResource(currentImgRes)
+
+    }
+
+    companion object {
+        const val ROTATION_DEGREES = 90F
+        val IMAGE_LIST = listOf(R.drawable.cat, R.drawable.dog, R.drawable.parrot)
     }
 }
```
Comment (PetrGrebnev) [inline app/src/main/java/com/example/lessson17/MainActivity.kt:143]: Approve

**Comment by IlyaSibogatov** [comment]
Approve

**Review by PetrGrebnev — APPROVED** [review]

File: app/src/main/java/com/example/lessson17/MainActivity.kt [hunk 8]
```diff
@@ -1,11 +1,140 @@
...
 import androidx.appcompat.app.AppCompatActivity
+import com.example.lessson17.databinding.ActivityMainBinding
+import java.text.DateFormat
+import java.time.LocalDateTime
+import java.time.format.DateTimeFormatter
+import java.util.*
+import kotlin.random.Random
 
 class MainActivity : AppCompatActivity() {
+    lateinit var binding: ActivityMainBinding
```
Comment (vshat-tms) [inline app/src/main/java/com/example/lessson17/MainActivity.kt]: private

File: app/src/main/java/com/example/lessson17/MainActivity.kt [hunk 9]
```diff
@@ -1,11 +1,140 @@
...
+    private var currentImgRes = R.drawable.cat
+    private var counter = 0
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
+        binding = ActivityMainBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+
```
Comment (vshat-tms) [inline app/src/main/java/com/example/lessson17/MainActivity.kt:31]: many empty lines

File: app/src/main/java/com/example/lessson17/MainActivity.kt [hunk 10]
```diff
@@ -1,11 +1,140 @@
...
+                info.text = currentTime
+            }
+
+            buttonToast.setOnClickListener {
+                Toast.makeText(this@MainActivity, "Hello from Stepan",
+                    Toast.LENGTH_SHORT).show()
+            }
+
+            buttonCat.setOnClickListener {
+                setImg(image, 0)
```
Comment (vshat-tms) [inline app/src/main/java/com/example/lessson17/MainActivity.kt]: 0,1,2,3 to const

File: app/src/main/java/com/example/lessson17/MainActivity.kt [hunk 11]
```diff
@@ -1,11 +1,140 @@
...
+        val img = (view as ImageView)
+        img.rotation = 0F
+        currentImgRes = IMAGE_LIST[imgResId]
+        img.setImageResource(currentImgRes)
+
+    }
+
+    companion object {
+        const val ROTATION_DEGREES = 90F
+        val IMAGE_LIST = listOf(R.drawable.cat, R.drawable.dog, R.drawable.parrot)
```
Comment (vshat-tms) [inline app/src/main/java/com/example/lessson17/MainActivity.kt:136]: val IMAGE_LIST_CAT_INDEX = 0
...

**Review by AverburhSiarhei — APPROVED** [review]

**Review by AverburhSiarhei — APPROVED** [review]

**Pull request closed without merging**